cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/DefaultScheduleBuilder.java

```diff
@@ -143,8 +144,8 @@ public ScheduleCreationSpec triggerOnProgramStatus(String programNamespace, Stri
                                                      ProgramStatus... programStatuses) {
     return new ScheduleCreationSpec(name, description, programName, properties,
                                     new ProgramStatusTrigger(
-                                      new ApplicationId(application, appVersion)
```

Comment:
With the introduction of `TriggerFactory`, can the `DefaultTriggerFactory` be used here to cleanup the code and not have duplicate trigger declarations?